File: src/client/VsClk.SignalService/ts/signalr-client-proxy/src/RelayHubMessageProperties.ts

export class RelayHubMessageProperties {
    
    public static propertySequenceId = 'sequenceId';

    public static propertyAuditPrefixId = 'audit-';

    public static propertyAuditPerfTypeId = 'perfType';

    public static createMessageSequence(sequence: number): { [key: string]: any; } {
        return {
            sequenceId: sequence,
        };
    }

    public static getMessageSequence(properties: { [key: string]: any; }): number | undefined  {
        if (properties && properties.hasOwnProperty(RelayHubMessageProperties.propertySequenceId)) {
            return Number(properties[RelayHubMessageProperties.propertySequenceId]);
        }
    }
}

File: src/services/containers/VsClk.SignalService/RelayServiceHub.Contracts/RelayHubMessageProperties.cs
// <copyright file="RelayHubMessageProperties.cs" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

using System.Collections.Generic;

namespace Microsoft.VsCloudKernel.SignalService
{
    /// <summary>
    /// Factory relay hub message properties.
    /// </summary>
    public static class RelayHubMessageProperties
    {
        /// <summary>
        /// The message property unique id.
        /// </summary>
        public const string PropertySequenceId = "sequenceId";

        /// <summary>
        /// The audit property prefix id.
        /// </summary>
        public const string PropertyAuditPrefixId = "audit-";

        /// <summary>
        /// The target property  prefix id.
        /// </summary>
        public const string PropertyTargetPrefixId = "target-";

        /// <summary>
        /// Indicate the special 'data' property.
        /// </summary>
        public const string PropertyDataId = "data";

        /// <summary>
        /// Create message properties based on a sequence id.
        /// </summary>
        /// <param name="sequenceId">The next sequence id.</param>
        /// <returns>A message properties dictionary.</returns>
        public static Dictionary<string, object> CreateMessageSequence(int sequenceId)
        {
            return new Dictionary<string, object>()
            {
                { PropertySequenceId, sequenceId },
            };
        }
    }
}
